### PR TITLE
feat: make the maximum recycleable parser cache size configurable

### PR DESCRIPTION
**Pull request by jfroundjian**
## Motivation and Context
https://github.com/wundergraph/cosmo/issues/1156

## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: router/pkg/config/fixtures/full.yaml [hunk 1]
```diff
@@ -212,6 +212,7 @@ engine:
   epoll_kqueue_conn_buffer_size: 128
   websocket_read_timeout: "5s"
   execution_plan_cache_size: 1024
+  resolver_max_recyclable_parser_size: 32768
```
Comment (jensneuse) [inline router/pkg/config/fixtures/full.yaml:215]: Where is this number coming from?

File: router/pkg/config/config.go [hunk 2]
```diff
@@ -227,6 +227,7 @@ type EngineExecutionConfiguration struct {
 	ParseKitPoolSize                       int                      `envDefault:"16" env:"ENGINE_PARSEKIT_POOL_SIZE" yaml:"parsekit_pool_size,omitempty"`
 	EnableValidationCache                  bool                     `envDefault:"true" env:"ENGINE_ENABLE_VALIDATION_CACHE" yaml:"enable_validation_cache"`
 	ValidationCacheSize                    int64                    `envDefault:"1024" env:"ENGINE_VALIDATION_CACHE_SIZE" yaml:"validation_cache_size,omitempty"`
+	ResolverMaxRecyclableParserSize        int                      `envDefault:"0" env:"ENGINE_RESOLVER_MAX_RECYCLABLE_PARSER_SIZE" yaml:"resolver_max_recyclable_parser_size,omitempty"`
```
Comment (jensneuse) [inline router/pkg/config/config.go:230]: I don't think we should keep this to be 0.

**Review by jensneuse — CHANGES_REQUESTED** [review]

File: router/pkg/config/fixtures/full.yaml (hunk 1 above)
Comment (jfroundjian) [reply router/pkg/config/fixtures/full.yaml:215]: This is the number of cached json values being parsed by the Parser. The Parser itself doubles the size of the underlying array every time it reaches the limit so a power of 2 is usually a good threshold imho.

File: router/pkg/config/config.schema.json [hunk 3]
```diff
@@ -1550,6 +1550,11 @@
           "type": "integer",
           "default": 1024,
           "description": "The size of the validation cache."
+        },
+        "resolver_max_recyclable_parser_size": {
```
Comment (jensneuse) [inline router/pkg/config/config.schema.json:1554]: Can you move this below the "parsekit_pool_size" item?

**Review by jensneuse — CHANGES_REQUESTED** [review]

**Review by jensneuse — APPROVED** [review]

**Pull request merged**